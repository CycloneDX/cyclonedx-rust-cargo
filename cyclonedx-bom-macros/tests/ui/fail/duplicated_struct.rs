use cyclonedx_bom_macros::versioned;

#[versioned("1.0", "2.0")]
mod base {
    // Define `Foo` for 1.0 twice
    #[versioned("1.0")]
    pub struct Foo;

    #[versioned("1.0")]
    pub struct Foo;
}

fn main() {
    let _ = v1_0::Foo;
}
