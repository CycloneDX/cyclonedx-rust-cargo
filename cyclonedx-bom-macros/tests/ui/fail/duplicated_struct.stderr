error[E0428]: the name `Foo` is defined multiple times
  --> $DIR/duplicated_struct.rs:10:5
   |
7  |     pub struct Foo;
   |     --------------- previous definition of the type `Foo` here
...
10 |     pub struct Foo;
   |     ^^^^^^^^^^^^^^^ `Foo` redefined here
   |
   = note: `Foo` must be defined only once in the type namespace of this module
